plugins/Proxy/Controller.php

```diff
@@ -26,6 +27,15 @@ class Controller extends \Piwik\Plugin\Controller
     const JS_MIME_TYPE = "application/javascript; charset=UTF-8";
 
     /**
+     * Proxy method to call API methods using current session for authentication
+     */
+    public function callApi()
+    {
+        echo FrontController::getInstance()->dispatch('API', 'index');
```

Comment:
Dispatching might return the content. `echo` is used at other places as well. See https://github.com/piwik/piwik/blob/master/plugins/Login/Login.php#L60 for example.